etw/cs/traceloggingdynamic/TraceLoggingDynamic.cs

```diff
@@ -873,43 +934,80 @@ public void AddUnicodeStringArray(string name, String[] values, EventOutType out
         /// </summary>
         public void AddAnsiString(string name, Byte[] value, EventOutType outType = EventOutType.Default, int tag = 0)
         {
-            this.AddScalarMetadata(name, InType.AnsiString, outType, tag);
+            this.AddMetadata(name, InType.AnsiString, outType, tag);
             this.AddScalarDataNulTerminatedByteString(value, 0, value.Length);
         }
 
+        /// <summary>
+        /// Adds an AnsiString field (nul-terminated MBCS).
+        /// NOTE: Prefer AddCountedAnsiString. Use AddAnsiString only if the decoder requires nul-terminated strings.
+        /// Meaningful outTypes: Default (String), Utf8, Xml, Json.
+        /// </summary>
+        public void AddAnsiString(string name, String value, Encoding encoding, EventOutType outType = EventOutType.Default, int tag = 0)

```

Comment:
Note: New overloads for all AnsiString methods. Old overloads expected AnsiString value to be `byte[]`, i.e. the user has to encode the string into a byte array. New overloads accept string + Encoding and we'll do the encoding for you.